conjur-oss/values.schema.json

```diff
@@ -0,0 +1,234 @@
+              "type": "string",
+              "minLength": 1
+            },
+            "expiration": {
+              "type": "integer",
+              "minimum": 1
+            }
+          }
+        },
+        "url": {
+          "type": "string"
+        }
+      }
+    },
+    "dataKey": {
+      "type": "string",
+      "minLength": 1
+    },
+    "deployment": {
+      "properties": {
+        "annotations": {
+          "type": "object"
+        }
+      }
+    },
+    "image": {
+      "properties": {
+        "pullPolicy": {
+          "type": "string"
+        },
+        "repository": {
+          "type": "string"
+        },
+        "tag": {
+          "type": "number"
+        }
+      }
+    },
+    "nginx": {
+      "properties": {
+        "image": {
+          "properties": {
+            "pullPolicy": {
+              "type": "string"
+            },
+            "repository": {
+              "type": "string"
+            },
+            "tag": {
+              "type": "string"
+            }
+          }
+        }
+      }
+    },
+    "nodeSelector": {
+      "type": "object"
+    },
+    "postgres": {
+      "properties": {
+        "image": {
+          "properties": {
+            "pullPolicy": {
+              "type": "string"
+            },
+            "repository": {
+              "type": "string"
+            },
+            "tag": {
+              "type": "string"
+            }
+          }
+        },
+        "persistentVolume": {
+          "properties": {
+            "create": {
+              "type": "boolean"
+            },
+            "size": {
+              "type": "string"
+            },
+            "storageClass": {
+              "type": "string"
+            }
+          }
+        }
+      }
+    },
+    "postgresLabels": {
+      "type": "object"
+    },
+    "rbac": {
+      "properties": {
+        "create": {
+          "type": "boolean"
+        }
+      }
+    },
+    "replicaCount": {
+      "type": "integer",
+      "minimum": 1
+    },
+    "resources": {
+      "type": "object"
+    },
+    "service": {
+      "properties": {
+        "external": {
+          "properties": {
+            "annotations": {
+              "type": "object"
+            },
+            "enabled": {
+              "type": "boolean"
+            },
+            "port": {
+              "type": "integer"
+            }
+          }
+        },
+        "internal": {
+          "properties": {
+            "annotations": {
+              "type": "object"
+            },
+            "port": {
+              "type": "integer"
+            },
+            "type": {
+              "type": "string"
+            }
+          }
+        }
+      }
+    },
+    "serviceAccount": {
+      "properties": {
+        "create": {
+          "type": "boolean"
+        },
+        "name": {
+          "type": "string"
+        }
+      }
+    },
+    "ssl": {
+      "required": [
+        "caKey",
```

Comment:
The `caKey`, `cert`, and `key` fields here should be conditionally required.
Thinking about this a bit, I believe the logic should be that `caKey`, `caCert`, and `key` are required if `cert` is set.

Again, I think you can use the `dependencies` keyword (see https://stackoverflow.com/questions/38717933/jsonschema-attribute-conditionally-required); e.g. add something like this (not sure if I have the syntax correct):
```
"dependencies": {
  "cert": { "required": ["key", "caCert", "caKey"] }
}
```
at line 205.